### PR TITLE
Upgrade DSI to acquire sub-daily granularity options

### PR DESCRIPTION
**Pull request by courtneyholcomb**
Primarily, this PR upgrades DSI. This is needed to get sub-daily granularity options into MF. Along the way:
- Points to the latest commit of dbt-core, which is the first commit that does not have a dependency conflict with the version of DSI we need.
- Updates various places in code affected by the upgrade.

### CODE REVIEW

File: dbt-metricflow/extra-hatch-configuration/requirements-cli.txt
Comment (tlento) [inline dbt-metricflow/extra-hatch-configuration/requirements-cli.txt:2]: Argh, this means we cannot release dbt-metricflow until 1.9 rolls out. I'm so frustrated.

File: metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py
Comment (tlento) [inline metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py:26]: Eh, maybe we just drop support for nanosecond for now, although I guess for cumulative metrics this isn't a big deal.

Note: BigQuery, Redshift, and Postgres only support microsecond. Trino and DuckDB support nanosecond although DuckDB uses a different type for it (and Trino requires a more explicit precision for anything smaller than millisecond).

File: metricflow-semantics/metricflow_semantics/time/time_constants.py
Comment (tlento) [inline metricflow-semantics/metricflow_semantics/time/time_constants.py:5]: Oh we're so close to removing this file....

**Review by tlento — COMMENTED** [review]
This whole dependency situation is a huge bummer.

Options:

1. We downgrade DSI to 0.5.2dev or whatever
2. We release MetricFlow and dbt-metricflow before merging it, and acknowledge that it's the only release of MetricFlow we'll deploy for public use prior to dbt-core 1.9
3. We figure something else out

I'm inclined towards option 2.

File: metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py
Comment (courtneyholcomb) [reply metricflow-semantics/metricflow_semantics/time/dateutil_adjuster.py:26]: As discussed offline, we'll keep as is.

**Comment by courtneyholcomb** [comment]
I believe this PR is only blocked due to waiting on `metricflow` / `dbt-metricflow` release, which will then allow us to point to a commit for `dbt-core`.

**Review by tlento — APPROVED** [review]

**Pull request merged**
